### ISSUE TITLE
Does not work on Xcode 8.1

### ISSUE THREAD

**Issue by harryworld** [fpg1503/MakeXcodeGr8Again#24]
Error log

```
/Applications/XcodeGr8.app/Contents/MacOS/Xcode - /Applications/XcodeGr8.app/Contents/MacOS/Xcode.unsignedreading infile: /Applications/XcodeGr8.app/Contents/MacOS/Xcode
Assertion failed: (dataoff != 0), function macho_unsign, file /Users/harryng/Documents/workspace/osx/opensource/MakeXcodeGr8Again/MakeXcodeGr8Again/Unsign/unsign.m, line 176.
(lldb) 
```

**Comment by iCodist** [comment]
+1

**Comment by podkovyrin** [comment]
Everything works fine on release build `Version 8.1 (8B62)`

**Comment by harryworld** [comment]
I'm on `Xcode 8.1 8B62`, `macOS 10.12.1 (16B2657)` and `MacBook Pro (Retina, Mid 2012)`
It does not work, when I drag Xcode into this

**Comment by rad182** [comment]
+1

**Comment by fpg1503** [comment]
You're dragging your `XcodeGr8.app`, this error normally occurs when the Xcode is already great, did you try installing any plugins? (This assertion failure means: the precondition of Xcode not being failed)

**Comment by rad182** [comment]
@fpg1503 Tried to delete plugins still crashing. BTW I'm not draggint eh XcodeGr8.app but the Xcode.app (updated from appstore)

**Comment by darktears** [comment]
Same here....

**Comment by MKGitHub** [comment]
Tried dragging Xcode 8.1 into the XcodeGr8.app window but it crashes!

**Comment by podkovyrin** [comment]
In my case, I downloaded Xcode archive from Apple Developer not from the AppStore

**Comment by davejewell** [comment]
Version 8.1 (8B62) not working for me.  I run XcodeGr8, patch 8.1 and launch it.  All looks good.  But as soon as I try to load an existing project into Xcode, the IDE hangs and I have to force-quit.  This is very repeatable.  Have tried it on earlier builds - same result.  

I am still running El Capitan.  Maybe this is the problem?  Are other people running Sierra with success?  I actually tried all this myself two or three months ago, but rather than SteakKnife, I was using the macho-unsign.py script.  This has the advantage that the unsigning can be restored: i.e. you can essentially resign Xcode because the original code signature is never discarded: it's just disabled.  

Regardless of technique used, I haven't been able to reliably unsign Xcode since 8.x was introduced.  A shame: would be good to figure out what the problem is.  Dave

**Comment by harryworld** [comment]
Experiment shows that the unsign works for Xcode 8.1 downloaded from Archive, the one on the left side. They both are `Xcode 8.1 (8B62)`

<img width="559" alt="Xcode 8.1" src="https://cloud.githubusercontent.com/assets/1784995/19828589/d12aa4b4-9dfc-11e6-9b22-cd883b65a8d9.png">

**Comment by davejewell** [comment]
Thanks Harry.  Further to my previous comments, I have got Version 8.1 (8B62) working - after a fashion (see below).  Last night, I tried podkovyrin's suggestion of downloading directly from Apple Developer rather than AppStore.  This made no difference for me. 
 
To be more specific, I can run the IDE, use my old plugins, build projects, etc >>PROVIDED<< that I take great care not to do anything that would invoke the standard open/save dialogs (NSOpenPanel, etc) from within the IDE.  If I try to open a project using File|Open..., then I get a hang.  But if I double-click the same project from the IDE, it loads up just fine. The call to NSOpenPanel is consistent with the Xcode .diag dumps in /Library/Logs/ DiagnosticReports.   Deeply strange.

**Comment by timothycosta** [comment]
With Xcode 8.1 I was not able to get XcodeGr8.app to run.  I'm now using the solution from Xvim and my plugins are working again.

https://github.com/XVimProject/XVim/blob/master/INSTALL_Xcode8.md

**Comment by davejewell** [comment]
Thanks Tim - will check out the XVim solution.

**Comment by davejewell** [comment]
Gosh - this is fun!  I tried creating a specific certificate exactly as per the XVim instructions.  All went well until codesign burped and said:

"SecKey API returned: -25293, (null)/Applications/Xcode.app: unknown error -1=ffffffffffffffff"

Despite this, Xcode ran fine but - as before - refused to load my plugins. 

At this point, I decided to go for broke.  I ran codesign again, but tried signing Xcode with my own, existing Mac Developer Application authority, as issued by Apple.  Somehow, I didn't expect this to work, but codesign seemed happy, and reported no error.  I can now load my existing plugins, and Xcode is rock-solid.  Kudos to Timothy and the XVim guys...

**Comment by fpg1503** [comment]
Woah! That's really awesome, I'll try to automate the XVim solution and add it to MakeXcodeGr8Again since it's working so well!

**Comment by MKGitHub** [comment]
Any update already?

**Comment by masuhara** [comment]
Same here. I can't use this for Xcode8.1. It crashes when Xcode was dragged and dropped in it.

**Comment by davejewell** [comment]
I'd *STRONGLY* encourage people to try the XVim technique discussed earlier.  It's fast, it's reliable and - at the end of the day - it's using Apple's own development tools to do the job.  Whether it's using those tools in a manner approved by Apple is less certain!  <grin>

I'm currently using Xcode 8.2 beta (8C30a) and it works just fine.

**Comment by DQ92** [comment]
I have just used XVim technique on Xcode 8.1 on elCapitan and unfortunately plugins are still invisible...

**Comment by davejewell** [comment]
OK - the usual questions: have you double checked that:

a) Plugins are in the correct place.  Eg:
~/Library/Application Support/Developer/Shared/Xcode/Plug-ins/XcodeExplorer.xcplugin 

b) Checked that you've updated info.plist in your plugin to have a UUID that's compatible with Xcode 8.1?  I don't have 8.1 installed right now (I'm using 8.2) otherwise I'd give you the UUID but basically just copy the DVTPlugInCompatibilityUUID key from Xcode's own info.plist.

If none of the above, then did you get an error reported when you used the XVim technique to re-sign Xcode?  Have you looked in the Console log for any clues there?

Dave

**Comment by DQ92** [comment]
Now, I used XVim on Xcode 8.2 beta on Sierra and no changes. Any errors in console, but Package Manager is still invisible in menu.
In /Library/Application Support/Developer/Shared/Xcode/Plug-ins/ I haven't XcodeExplorer.xcplugin...  DVTPlugInCompatibilityUUID key is correct.

**Comment by davejewell** [comment]
Personally, I wouldn't start with Package Manager (I assume you're talking about Alcatraz) because it's a relatively complex plugin.  There's potentially a lot to go wrong when moving to a new version of Xcode.  If I were you, I'd try a relatively simple plugin like XcodeExplorer.  Just compile it "as-is" without making any changes, update the UUID for Xcode 8.2 and it should just work.  

If you still have problems, go to ~/Library/Preferences, open the file com.apple.dt.Xcode.plist and have a look in this file.  Tell me what you see in the section that I've expanded out in my screenshot (attached).  

Dave

![temp](https://cloud.githubusercontent.com/assets/823898/20755134/c1e8a2e0-b705-11e6-9006-8fadf35b3e47.png)

**Comment by DQ92** [comment]
I did the following steps:

Version 8.2 beta (8C30a)

1. Remove all Xcodes from:
/Application
/Application Shared
/Developer
/Preferences

2. Remove Alcatraz
3. Restart mac
4. Copy Xcode to Applications (Xcode: 12 330 151 465 B (12,94 GB))
5. Create certificate
6. sudo codesign -f -s XcodeSigner /Applications/Xcode.app

/Applications/Xcode.app: replacing existing signature
 
You have not agreed to the Xcode license agreements. You must agree to both license agreements below in order to use Xcode.

Hit the Enter key to view the license agreements at '/Applications/Xcode.app/Contents/Resources/English.lproj/License.rtf'

Apple Inc.

Xcode and Apple SDKs Agreement

.
.
.

By typing 'agree' you are agreeing to the terms of the software license agreements. Type 'print' to print them or anything else to cancel, [agree, print, cancel] 
You can view the license agreements in Xcode's About Box, or at /Applications/Xcode.app/Contents/Resources/English.lproj/License.rtf

/Applications/Xcode.app: the codesign_allocate helper tool cannot be found or used

7. Open Xcode, Agree terms and close Xcode. Again:
sudo codesign -f -s XcodeSigner /Applications/Xcode.app

8. $ xcode-select -p
/Applications/Xcode.app/Contents/Developer

9. XVim-master daniel$ sudo make
** BUILD SUCCEEDED **

10. Launch Xcode

11. Install Alcatraz
curl -fsSL https://raw.github.com/alcatraz/Alcatraz/master/Scripts/install.sh | sh

-----
TEST

1. Library/Application Support/Developer/Shared/Xcode/Plug-ins
ls -a
.				
AutoIndentWithSave.xcplugin	
DefaultMarginDisabler.xcplugin	
Refactorator.xcplugin		
XcodeColors.xcplugin
..				
CocoaPodUI.xcplugin		
FixCode.xcplugin		
SwiftJsonToObj.xcplugin
Alcatraz.xcplugin		
CocoaPods.xcplugin		
GitDiff.xcplugin		
XCSwiftr.xcplugin

2.  ~/Library/Preferences, open the file com.apple.dt.Xcode.plist 

(screen)

Xcode: 12 357 566 024 B (12,97 GB)

![zrzut ekranu 2016-12-01 o 23 45 44](https://cloud.githubusercontent.com/assets/11273569/20816231/9615de90-b821-11e6-8310-0b693126befb.png)

**Comment by davejewell** [comment]
Hmmm... at this point, all I can do is highlight the differences between your methodology and mine.  Hopefully, something is significant:

1) I am running Sierra 10.12.1 (16B2657) not El Capitan, or Yosemite.
2) Where you say "Create certificate", I think this is maybe the most significant point.  As I said 3 weeks back, I found that creating a certificate specifically for the purpose of resigning Xcode did not work for me.  What DID work was to use my existing Mac Developer certificate, issued by Apple.  If you are a Mac developer, you should maybe try that.  
3) XVim-master daniel$ sudo make.  It should not be necessary to build XVim.  The only point is that the XVim guys came up with this technique for re-signing Xcode.
4) Install Alcatraz.  Like I said, forget about Alcatraz for now and try something simpler.  Even if Xcode was not signed, I have no either whether Alcatraz would work with 8.2.  I do know that Xcode-Explorer works fine with 8.2.  See my accompanying screenshot: this shows Xcode-Explorer running inside 8.2, displaying a small extension to the plugin which I wrote.
5) When you talk about "Library/Application Support/Developer/Shared/Xcode/Plug-ins", hopefully you mean: "~/Library/Application Support/Developer/Shared/Xcode/Plug-ins"  :-)   You should clear out this directory!  How can you expect to find the problem if you have nearly a dozen plugins in there??? 
6) Looking at the plist dump you supplied, I can't see references to DVTPlugInManagerNonApplePlugIns.  It's as if Xcode isn't seeing any of your plugins at all.  

Bottom line: clear out the plugin directory, test with just *ONE* simple plugin which is know to work, re-sign Xcode using your developer certificate, (which of course is signed by Apple).  Also, don't simply use curl to install an old binary for your plugin.  Download the Xcode-Explorer source and build it from scratch using 8.2.  

If you still can't make it work, I can tell you how to activate verbose logging options for Xcode which will hopefully give us more of a handle on what is going wrong.  

Dave

![xcode-explorer](https://cloud.githubusercontent.com/assets/823898/20817299/b0ad4804-b81e-11e6-9b6f-b622cf656bd7.png)

**Comment by DQ92** [comment]
Ok I update my mac to Sierra and I used my Mac Developer certificate and it's work! In Xcode I have access to my plugins. Thank you very much davejewell for your time to me.

**Comment by davejewell** [comment]
Yay!  That's great Daniel.  Delighted that you got it working.  :-)

**Comment by fpg1503** [comment]
Have you tried using the [non-GUI version](https://github.com/fpg1503/MakeXcodeGr8Again#non-gui-version)?
